### PR TITLE
Remove panic! from the compositor

### LINKED ISSUE #10864: Remove panic! from compositor
We got rid of the uses of `.unwrap()` but not `panic!`.

### PR DESCRIPTION
**Pull request by asajeffrey**
Fixes #10864, and adds a check to `/etc/ci/check_no_unwrap.sh`.

r? @aneeshusa

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10902)

### CODE REVIEW

File: etc/ci/check_no_unwrap.sh [hunk 1]
```diff
@@ -12,4 +12,4 @@ FILES=("components/compositing/compositor.rs"
        "components/compositing/pipeline.rs"
        "components/compositing/constellation.rs")
 
-! grep -n "unwrap(" "${FILES[@]}"
+! grep -n "unwrap(\|panic!" "${FILES[@]}"
```
Comment (aneeshusa) [inline etc/ci/check_no_unwrap.sh]: Let's make this `panic!(` for consistency. LGTM otherwise.

File: components/compositing/compositor.rs [hunk 2]
```diff
@@ -1119,7 +1125,7 @@ impl<Window: WindowMethods> IOCompositor<Window> {
                                                    layer_id: LayerId) {
         if let Some(point) = self.fragment_point.take() {
             if !self.move_layer(pipeline_id, layer_id, Point2D::from_untyped(&point)) {
-                panic!("Compositor: Tried to scroll to fragment with unknown layer.");
+                warn!("Compositor: Tried to scroll to fragment with unknown layer.");
             }
```
Comment (samlh) [inline components/compositing/compositor.rs]: Return?

File: etc/ci/check_no_unwrap.sh (hunk 1 above)
Comment (asajeffrey) [inline etc/ci/check_no_unwrap.sh]: Done.

File: components/compositing/compositor.rs (hunk 2 above)
Comment (asajeffrey) [inline components/compositing/compositor.rs]: Done.

**Comment by aneeshusa** [comment]
r=me after squash

Once the buildbot config is part of this repo, I'd like to rename the check_no_unwrap.sh to check_no_panic.sh; I'll probably open that as an E-easy issue.

**Pull request merged**